contrib/monitored_resource_util/src/main/java/io/opencensus/contrib/monitoredresource/util/MonitoredResourceUtils.java

```diff
@@ -50,5 +53,29 @@ public static MonitoredResource getDefaultResource() {
     return null;
   }
 
+  /**
+   * Returns a {@code Resource}. Detector sequentially runs resource detection from environment
+   * variables, GKE, GCP and AWS.
+   *
+   * @return a {@code Resource}.
+   * @since 0.18
+   */
+  @Nullable
+  public static Resource detectResource() {
```

Comment:
You can also deprecate the other API `getDefaultResource` and the `MonitoredResource` class.

Reply:
```@deprecated``` getDefaultResource API.